.jenkins/Jenkinsfile

```diff
@@ -1,121 +1,202 @@
 // Copyright (c) Open Enclave SDK contributors.
 // Licensed under the MIT License.
 
-// Check if BRANCH is specified by multibranch pipeline job
-GLOBAL_BRANCH_NAME = env.BRANCH_NAME ?: params.BRANCH
-
-// Regex that includes directory you want to ignore for CI builds.
-String IGNORED_DIRS = "^docs|^\\.jenkins/infrastructure|\\.md\$|^VERSION\$|^OWNERS\$"
-
-// Load OpenEnclaveJenkinsLibrary version to use in this priority:
-//     1. If this is a bors run, use the bors branch
-//     2. Use params.OECI_LIB_VERSION if it is specified
-//     3. If none of the above, default to master
-if ( REPOSITORY_NAME == 'openenclave/openenclave' && GLOBAL_BRANCH_NAME ==~ /^(trying|staging)$/ ) {
-    GLOBAL_OECI_LIB_VERSION = GLOBAL_BRANCH_NAME
-} else if ( params.OECI_LIB_VERSION ) {
-    // Use regex to match bors branches to include any changes to OpenEnclaveJenkinsLibrary
-    GLOBAL_OECI_LIB_VERSION = params.OECI_LIB_VERSION
+/* Prevent Branch Indexing from triggering a build. This is necessary because
+   Branch Indexing will trigger a build for every Pull Request in the repository
+   every time it occurs and waste resources.
+*/
+build_cause = currentBuild.getBuildCauses().toString()
+if (build_cause.contains('BranchIndexingCause')) {
+  currentBuild.result = 'ABORTED'
+  error("Branch Indexing is not allowed. Please trigger manually or via a pull request.")
 } else {
-    GLOBAL_OECI_LIB_VERSION = "master"
+  println("Build cause: ${build_cause}")
 }
-library "OpenEnclaveJenkinsLibrary@${GLOBAL_OECI_LIB_VERSION}"
+
+/* A list of regex that includes directories and files to be ignored for CI builds.
+  This is used as a grep extended regular expression.
+  See https://www.gnu.org/software/grep/manual/html_node/Basic-vs-Extended.html
+*/
+List<String> LIST_IGNORED_DIRS = [
+    '^docs',
+    '^.jenkins/infrastructure',
+    '^.jenkins/docker',
+    '^.github',
+    '.md\$',
+    '^VERSION\$',
+    '^OWNERS\$'
+]
+
+// This joins list into regex string to be used in git diff
+String IGNORED_DIRS_REGEX = LIST_IGNORED_DIRS.join('|')
+
+List<String> APPROVED_AUTHORS = [
```

Comment:
Will this prevent building pull requests from contributors that are not in this list?

Reply:
For contributors not in this list, it will prevent builds from proceeding past the initial "Check access" stage. So, to test those pull requests, someone with the permission will need to start a build on Jenkins. This works similarly to bors where only certain people were able to run bors try / r+